File: data/code_pairs/fortran/478.f90
program array_char
  implicit none
  integer :: i
  integer, allocatable :: j(:)
  character(len=5) :: x, y
  character(len=5), allocatable :: z(:)
  allocate(j(5))
  allocate(z(2))
  
  x = "ab"
  y = "cd"
  z = [y(1:len(trim(y))) // x(1:len(trim(x))), x(1:len(trim(x))) // y(1:len(trim(y)))]
  
  j = [(ichar(z(1)(i:i)), i=1,5)]
  if (any(j /= [99,100,32,32,32])) stop 1
  
  j = [(ichar(z(2)(i:i)), i=1,5)]
  if (any(j /= [97,98,32,32,32])) stop 2
  
  x = "a "
  z = [y(1:len(trim(y))) // x(1:len(trim(x))), x(1:len(trim(x))) // y(1:len(trim(y)))]
  
  print *, "Fortran code executed successfully."
end program array_char

File: data/code_pairs/cpp/478.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

int main() {
    std::vector<int> j(5);
    std::string x = "ab";
    std::string y = "cd";
    std::vector<std::string> z(2);
    
    z[0] = y.substr(0, y.find_last_not_of(" ") + 1) + x.substr(0, x.find_last_not_of(" ") + 1);
    z[1] = x.substr(0, x.find_last_not_of(" ") + 1) + y.substr(0, y.find_last_not_of(" ") + 1);
    
    for (size_t i = 0; i < 5; ++i) {
        j[i] = (i < z[0].size()) ? static_cast<int>(z[0][i]) : 32; // Fill with ASCII of space if out of bounds
    }
    if (j != std::vector<int>{99, 100, 32, 32, 32}) {
        std::cerr << "Error 1" << std::endl;
        return 1; // Stop the program with an error
    }
    
    for (size_t i = 0; i < 5; ++i) {
        j[i] = (i < z[1].size()) ? static_cast<int>(z[1][i]) : 32; // Fill with ASCII of space if out of bounds
    }
    if (j != std::vector<int>{97, 98, 32, 32, 32}) {
        std::cerr << "Error 2" << std::endl;
        return 2; // Stop the program with an error
    }
    
    x = "a ";
    z[0] = y.substr(0, y.find_last_not_of(" ") + 1) + x.substr(0, x.find_last_not_of(" ") + 1);
    z[1] = x.substr(0, x.find_last_not_of(" ") + 1) + y.substr(0, y.find_last_not_of(" ") + 1);
    
    std::cout << "C++ code executed successfully." << std::endl;
    
    return 0; // Program completed successfully
}